app/src/main/java/agency/five/codebase/android/movieapp/data/di/DataModule.kt

```diff
@@ -1,12 +1,25 @@
 package agency.five.codebase.android.movieapp.data.di
 
-import agency.five.codebase.android.movieapp.data.repository.FakeMovieRepository
 import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.data.repository.MovieRepositoryImpl
 import kotlinx.coroutines.Dispatchers
+import org.koin.core.qualifier.named
 import org.koin.dsl.module
 
 val dataModule = module {
-    single<MovieRepository> {
-        FakeMovieRepository(ioDispatcher = Dispatchers.IO)
+
+    single<MovieRepository>(named("dispatcherIo")) {
+        MovieRepositoryImpl(
+            movieService = get(),
+            movieDao = get(),
+            bgDispatcher = Dispatchers.IO
+        )
+    }
+    single<MovieRepository>(named("dispatcherMain")) {
+        MovieRepositoryImpl(
+            movieService = get(),
+            movieDao = get(),
+            bgDispatcher = Dispatchers.Main
+        )
```

Comment:
Why do you have this? Remove it and remove named from the `MovieRepository` above it.